src/ride/track.c

```diff
@@ -370,7 +370,7 @@ void track_list_populate(ride_list_item item, uint8* track_list_cache){
 		}
 
 		// If cur_track_entry_index is greater than max number of tracks
-		if (cur_track_entry_index >= 1000){
+		if (cur_track_entry_index >= 254){
```

Comment:
1000 seems to be a arbitrary watch value, however, `cur_track_entry_index` is of type `uint8` (line 341) thus 254 is the highest sane value with which we can compare here.